ui/app/AppLayouts/Wallet/services/dapps/DAppsModule.qml

```diff
@@ -0,0 +1,309 @@
+        id: requests
+    }
+
+    ChainsSupervisorPlugin {
+        id: chainsSupervisorPlugin
+
+        sdk: root.wcSdk
+        networksModel: root.networksModel
+    }
+
+    Connections {
+        target: root.wcSdk
+        enabled: root.wcSdk.enabled
+
+        function onPairResponse(ok) {
+            root.pairingResponse(ok)
+        }
+    }
+
+    SiweRequestPlugin {
+        id: siwePlugin
+
+        readonly property var connectionRequests: new Map()
+        sdk: root.wcSdk
+        store: root.store
+        accountsModel: root.accountsModel
+        networksModel: root.networksModel
+
+        onRegisterSignRequest: (request) => {
+            requests.enqueue(request)
+        }
+
+        onUnregisterSignRequest: (requestId) => {
+            const request = requests.findById(requestId)
+            if (request === null) {
+                console.error("SiweRequestPlugin::onUnregisterSignRequest: Error finding event for requestId", requestId)
+                return
+            }
+            requests.removeRequest(request.topic, requestId)
+        }
+
+        onConnectDApp: (chains, dAppUrl, dAppName, dAppIcon, key) => {
+            siwePlugin.connectionRequests.set(key.toString(), {chains, dAppUrl, dAppName, dAppIcon})
+            root.connectDApp(chains, dAppUrl, dAppName, dAppIcon, key)
+        }
+
+        onSiweFailed: (id, error, topic) => {
+            root.siweCompleted(topic, id, error)
+        }
+
+        onSiweSuccessful: (id, topic) => {
+            d.lookupSession(topic, function(session) {
+                // Persist session
+                if(!root.store.addWalletConnectSession(JSON.stringify(session))) {
+                    console.error("Failed to persist session")
+                }
+
+                root.siweCompleted(topic, id, "")
+            })
+        }
+
+        function accept(key, approvedChainIds, accountAddress) {
+            const approvedNamespaces = JSON.parse(
+                DAppsHelpers.buildSupportedNamespaces(approvedChainIds,
+                                                [accountAddress],
+                                                SessionRequest.getSupportedMethods()))
+            siwePlugin.connectionApproved(key, approvedNamespaces)
+        }
+        function reject(key) {
+            siwePlugin.connectionRejected(key)
+        }
+    }
+
+    SQUtils.QObject {
+        id: d
+
+        function lookupSession(topicToLookup, callback) {
+            wcSdk.getActiveSessions((res) => {
+                Object.keys(res).forEach((topic) => {
+                    if (topic === topicToLookup) {
+                        let session = res[topic]
+                        callback(session)
+                    }
+                })
+            })
+        }
+    }
+
+    // bcSignRequestPlugin and wcSignRequestPlugin are used to handle sign requests
+    // Almost identical, and it's worth extracting in an inline component, but Qt5.15.2 doesn't support it
```

Comment:
For some reason having any inline components here will freeze the app on qt 5.15.2